File: 프로그래머스/2/42586. 기능개발/기능개발.cs
using System;
using System.Collections.Generic;

public class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        List<int> result = new List<int>();
        
        int[] completeday = new int[progresses.Length];
        
        for(int i = 0; i < progresses.Length; i++)
        {
            completeday[i] = (int)Math.Ceiling((100.0-progresses[i])/speeds[i]);
            
        }
        int maxDays = completeday[0]; 
        int count = 1;
        
        for (int i = 1; i < progresses.Length; i++)
        {
            if (completeday[i] <= maxDays)
            {
                count++; 
            }
            else
            {
                result.Add(count); 
                maxDays = completeday[i];count = 1; 
            }
        }
        result.Add(count);
        
        
        return result.ToArray();
    }
}

File: 프로그래머스/2/42586. 기능개발/기능개발.java
import java.util.*;

class Solution {
    public int[] solution(int[] progresses, int[] speeds) {
        Queue<Integer> queue = new LinkedList<>();
        List<Integer> result = new ArrayList<>();

        for(int i = 0; i < progresses.length; i++) 
        {
            int days = (int)Math.ceil((100.0 - progresses[i]) / speeds[i]);
            queue.offer(days);
        }
        while(!queue.isEmpty()) 
        {
            int current = queue.poll(); 
            int count = 1;

            while(!queue.isEmpty() && queue.peek() <= current) 
            {
                queue.poll();
                count++;
            }
            result.add(count); 
        }
        return result.stream().mapToInt(i -> i).toArray();
    }
}